scio-test/src/main/scala/com/spotify/scio/testing/SCollectionMatchers.scala

```diff
@@ -91,16 +91,24 @@ trait SCollectionMatchers {
 
   private def makeFn[T](f: JIterable[T] => Unit): SerializableFunction[JIterable[T], Void] =
     new SerializableFunction[JIterable[T], Void] {
+      // delegate serialization to Kryo to avoid serialization issues in tests
+      // when a non-serializable object is captured by the closure
+      val impl = Externalizer(f)
```

Comment:
yeah java serde is first (kryo is really a safety net)... We would need to have some examples ... but I think this could™ work well in conjunction with `ClosureCleaner`?

Reply:
I verified that `Externalizer` actually falls back to kryo for the lambda and works, different from what I remembered.

Reply:
@nevillelyh Yeah the test fails without it so it does something for sure :)
Should we merge the PR then ?

Reply:
Yeah